### DISCUSSION TITLE
Open Source Backends for Quasar SAAS?

### DISCUSSION THREAD

**Discussion by sonicviz** [quasarframework/quasar#15100] [CLI - PWA mode]
Any suggestions for robust backends that you could hook up to Quasar to jumpstart a SAAS?

Apart from Firebase. Looking for robust open source SAAS (ie: multi tenant) solutions that have the following (or some) of the core features:

- Authentication
- Security
- User and Role management
- Payments (Stripe preferred OOTB)
- Audit Logs
- Settings
- I18n
- Debugging

A couple I've looked at, with reasons:
- https://scaffoldhub.io/ Not updated for a while, Vue2/Vuex going to be discontinued (React only now). Nodejs 16 only.
- https://pocketbase.io/docs/ Very neat minimalist with a Quasar starter kit https://github.com/benallfree/awesome-pocketbase but could be a touch early for production use as it's under active development to V1 still. 
- https://strapi.io/ seems possible, but might be too heavyweight? Anyone with experience of this using it for a SAAS?
- WordPress headless, via Atlas https://github.com/wpengine/faustjs or similar

Would appreciate any other suggestions to seriously consider. Thanks!

Update: Found this thread after I posted this, has some useful info: https://forum.quasar-framework.org/topic/7852/which-backend-for-quasar/

**Comment by TobyMosque** [comment]
do u tried supabase?

**Comment by sonicviz** [comment]
Not yet, but will take a look. Thanks!

**Comment by sonicviz** [comment]
Ok, I checked out Supabase. Very nice overall, but unfortunately webhooks are still alpha and are not recommended for production yet. 
Webhooks are needed for implementing payments callbacks from Stripe etc,

Apart from that (showstopper atm) it looks great. I like the Auth/roles/permissions/policies setup, and the included email is really slick. 

Worth keeping an eye on for when Webhooks is production ready.

**Reply by TobyMosque** [reply]
personally, in your place, i would take a more common framework:
- https://nestjs.com
- https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-7.0&tabs=visual-studio
- https://tsed.io
- https://feathersjs.com

**Comment by sonicviz** [comment]
Thanks for the suggestions.

I think Strapi does what I need, though I don't need all the CMS part really. Will see after I play with it a bit more. 
Tricky stuff choosing a backend to integrate.
